File: Javascript/DesignParkingSystem.js
// Leetcode Problem - 1603

var ParkingSystem = function(big,small,medium){
    this.big = big
    this.bigCur = 0
    this.medium = medium
    this.mediumCur = 0
    this.small = small
    this.smallCur = 0
}

ParkingSystem.prototype.addCar = function(carType){
    if (carType === 1){
        if (this.bigCur<this.big){
            this.bigCur+=1
            return true
        }
        else{
            return false
        }
    }else if (carType === 2){
        if (this.mediumCur<this.medium){
            this.mediumCur+=1
            return true
        }else{
            return false
        }
    }else if (carType === 3){
        if (this.smallCur<this.small){
            this.smallCur+=1
            return true
        }
        else{
            return false
        }
    }
}

File: Python/DesignParkingSystem.py
## Leetcode Problem - 1603

class ParkingSystem:

    def __init__(self,big,medium,small) -> None:
        self.big = big
        self.bigCur = 0
        self.medium = medium
        self.mediumCur = 0
        self.small = small
        self.smallCur = 0

    def addCar(self,carType):
        if carType == 1:
            if self.bigCur<self.big:
                self.bigCur+=1
                return True
            else:
                return False
        elif carType == 2:
            if self.mediumCur<self.medium:
                self.mediumCur+=1
                return True
            else:
                return False
        elif carType == 3:
            if self.smallCur<self.small:
                self.smallCur+=1
                return True
            else:
                return False
